src/backends/interflop-mca-int/interflop_mca_int.c

```diff
@@ -165,30 +166,23 @@ static __thread rng_state_t rng_state;
 /* -126-24+-126-24 = -300 > DOUBLE_EXP_MIN (-1022) */
 static inline void _noise_binary64(double *x, const int exp,
                                    rng_state_t *rng_state) {
-  int64_t noise;
-  uint32_t shift;
-
   // Convert preserving-bytes double to int64_t
-  int64_t x_s64 = *(int64_t *)(x);
+  binary64 *b64 = (binary64 *)x;
 
   // amount by which to shift the noise term sign (1) + exp (11) + noise
   // exponent
-  shift = 1 + DOUBLE_EXP_SIZE - exp;
+  const uint32_t shift = 1 + DOUBLE_EXP_SIZE - exp;
 
   // noise is a signed integer so the noise is centered around 0
   // only 32 bits of noise are used, they are left aligned in a signed 64 bit
-  noise = ((uint64_t)_get_rand_uint32(rng_state, &global_tid_lock, &global_tid)
-           << 32);
+  int64_t noise = get_rand_uint64(rng_state, &global_tid) & 0xFFFFFFFF00000000;
```

Comment:
Remove the mask here. Better to use the 32 extra random bits since we paid for them :-)

The code before was just a trick to avoid using 64 random bits; using 32 gave similar results and was cheaper.